biz.aQute.bndlib.tests/test/test/annotationheaders/SPIAnnotationsTest.java

```diff
@@ -484,6 +484,38 @@ public void testServiceProvider_serviceProperties() throws Exception {
 		}
 	}
 
+	@Test
+	public void testServiceProviderProvidesService() throws Exception {
+		try (Builder b = new Builder();) {
+			b.addClasspath(IO.getFile("bin_test"));
+			b.setPrivatePackage("test.annotationheaders.spi.provider");
+			b.build();
+			b.getJar()
+				.getManifest()
+				.write(System.out);
+			assertTrue(b.check());
+
+			Attributes mainAttributes = b.getJar()
+				.getManifest()
+				.getMainAttributes();
+
+			Header req = Header.parseHeader(mainAttributes.getValue(Constants.REQUIRE_CAPABILITY));
+			assertEquals(2, req.size());
+
+			assertExtender(req, "osgi.serviceloader.registrar");
+			assertEE(req);
+
+			Header cap = Header.parseHeader(mainAttributes.getValue(Constants.PROVIDE_CAPABILITY));
+			assertEquals(3, cap.size());
+
+			Props p = cap.get("osgi.service");
+			assertNotNull(p);
+			assertNotNull(p.get("objectClass:List<String>"));
+			assertEquals("test.annotationheaders.spi.SPIService", p.get("objectClass:List<String>"));
```

Comment:
Shouldn't OSGiHeader parse this into Parameters where the `List<String>`-ness if parsed out into the Attrs?

Reply:
will do!